handlers/handlers.go

```diff
@@ -194,10 +195,131 @@ func (s *ServiceSet) serviceHTML(db *mbtiles.DB) handlerFunc {
 	}
 }
 
+type tileCoord struct {
+	z    uint8
+	x, y uint64
+}
+
+// extractTileCoord extracts and returns tile coordinates and an optional
+// extension from a byte slice and returns them. In case the byte slice is too
+// short or the number parsing cannot be performed, the returned error is
+// non-nil.  pcs is expected to be in order (z, y, x) when the parameter swapXY
+// is false, and in order (z, x, y) when swapXY is true. The extension is
+// extracted from the third element of pcs. Any elements in pcs beyond the
+// third are ignored.
+func extractTileCoord(pcs []string, swapXY bool) (tc tileCoord, ext string, err error) {
+	if len(pcs) < 3 {
+		err = fmt.Errorf("not enough elements to extract tile coordinates")
+		return
```

Comment:
@fawick thank you for teaching me about [naked return statements](https://tour.golang.org/basics/7)!